packages/atomic/src/components/common/pager/pager-button.tsx

```diff
@@ -0,0 +1,77 @@
+import {FunctionalComponent, h} from '@stencil/core';
+import {Button, ButtonProps} from '../button';
+import {RadioButton, RadioButtonProps} from '../radio-button';
+
+export interface PagerNavigationButtonProps
+  extends Omit<ButtonProps, 'style' | 'part' | 'class'> {
+  icon: string;
+}
+
+export interface PagerPageButtonProps extends RadioButtonProps {
+  page: number;
+  isSelected: boolean;
+  text: string;
+}
+
+export const PagerPreviousButton: FunctionalComponent<
+  PagerNavigationButtonProps
+> = (props) => {
+  return (
+    <Button
+      {...props}
+      style="outline-primary"
+      part="previous-button"
+      class="p-1 min-w-[2.5rem] min-h-[2.5rem] flex justify-center items-center"
```

Comment:
In the functional components defined here, should we spread `props` after having declared the default props, to allow overriding them?
```suggestion
      style="outline-primary"
      part="previous-button"
      class="p-1 min-w-[2.5rem] min-h-[2.5rem] flex justify-center items-center"
      {...props}
```

Reply:
Since `PagerNavigationButtonProps` is an interface that `Omit<ButtonProps, 'style' | 'part' | 'class'>`, there's no override that should enter the equation.

So putting the prop spread before or after should have no effect !

Reply:
And moreover, we don't want to override the style or class from upstream.